enhancements/sdk-metrics.md

```diff
@@ -0,0 +1,120 @@
+  - "@joelanford"
+approvers:
+  - "@dmesser" 
+  - "@joelanford"
+creation-date: 2020-05-19
+last-updated: 2020-05-19
+status: implementable
+---
+
+# Operator SDK metrics
+
+## Release Signoff Checklist
+
+- [X] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+
+## Summary
+
+The enhancement is intended to define the bundle resources in which SDK labels can be added, so that they can be read by OLM and included in the current OLM metrics. These metrics will help us in providing a measure to evaluate the reachability of Operator SDK among community operators.
+
+## Motivation
+
+Collecting metrics regarding the usage of Operator SDK among operator authors is important to measure the reachability of the tool. Since OLM already has the pipeline built to obtain metrics on the operators and report it to the redhat telemetry, we intend to utilize the same to obtain data on the operators using SDK to build bundle images.
+
+### Goals
+
+The goal of this enhancement is to :
+1.  List the bundle resources on which SDK stamp will be added.
+2. The format of the stamp which will be added.
+
+### Design details
+
+The current bundle image has the following project structure:
+
+```
+ ...
+  bundle.Dockerfile
+  ...
+  $ tree ./deploy/olm-catalog/test-operator/
+  ./deploy/olm-catalog/test-operator/
+  ├── manifests
+  │   ├── example.com_tests_crd.yaml
+  │   └── test-operator.clusterserviceversion.yaml
+  └── metadata
+      └── annotations.yaml
+```
+
+SDK stamps would be present in the following locations:
+
+1. bundle.Dockerfile
+
+The dockerfile will include the sdk stamp in the form of `LABEL` which will follow the following format:
+
+```
+LABEL operators.operatorframework.io.bundle.sdk.v1 = operator_type:sdk_version
+```
+
+2. annotations.yaml
+
+Similar to the format followed in the dockerfile, `annotations.yaml` will have the following label:
+
+```
+operators.operatorframework.io.bundle.sdk.v1: operator_type:sdk_version
+```
+
+For example, in case of memcached-operator, the labels can be written as:
+
+```
+LABEL operators.operatorframework.io.bundle.sdk.v1 = go:v0.17.0
+```
+
+3. Cluster service version
+
+The cluster service version generated by SDK using the `operator-sdk generate csv` command can have a similar stamp included in the metadata. The stamp will be present as an [annotation][kubernetes_annotation]. For example, while generating a new CSV, the following will be added to Object metadata:
+
+```Go
+func newCSV(name, version string) (*olmapiv1alpha1.ClusterServiceVersion, error) {
+    csv := &olmapiv1alpha1.ClusterServiceVersion{
+        ...
+        ObjectMeta: metav1.ObjectMeta{
+			Name:      getCSVName(name, version),
+			Namespace: "placeholder",
+			Annotations: map[string]string{
+				"capabilities": "Basic Install",
+                "csv-builder": "sdk-go-v0.17.0",
```

Comment:
It would probably make sense to make this align more closely with the labels being inserted in other places. i.e.:
```go
Annotations: map[string]string{
	"capabilities": "Basic Install",
	"sdk.operatorframework.io/version": "v0.17.0",
	"sdk.operatorframework.io/layout": "go.kubebuilder.io/v2.0.0",
}

Reply:
If we version labels, we should version annotation keys too:
```
"sdk.operatorframework.io/version/v1": "v0.17.0",
```